flagging_site/blueprints/flagging.py

```diff
@@ -71,13 +72,38 @@ def output_model() -> str:
     
     returns: render model on output_model.html
     """
+
+    #parse contents of the query string to get reach & hours parameter
+    #default is  reach = 0 and hours = 24
+    reach = request.args.get('reach', 0)
+    hours = request.args.get('hours', 24)
+        
+    #convert to integers
+    hours = int(hours)
+    reach = int(reach)
+    
     df_hobolink = get_hobolink_data('code_for_boston_export_21d')
     df_usgs = get_usgs_data()
     df = process_data(df_hobolink, df_usgs)
-    table_html = '<br /><br />'.join(map(stylize_model_output, [
-        reach_2_model(df),
-        reach_3_model(df),
-        reach_4_model(df),
-        reach_5_model(df)
-    ]))
+    
+    reach_model_mapping = {
+        2: reach_2_model,
+        3: reach_3_model,
+        4: reach_4_model,
+        5: reach_5_model
+    }
+    
+    if reach in reach_model_mapping:
+        reach_func = reach_model_mapping[reach]
+        reach_list = [reach_func(df, rows = hours)]
+    else:
+        reach_list = [
+            reach_2_model(df, rows = hours),
+            reach_3_model(df, rows = hours),
+            reach_4_model(df, rows = hours),
+            reach_5_model(df, rows = hours)
+        ]
```

Comment:
Excellent stuff; very Pythonic and efficient way to do this.

Make sure to remove the space between the kwarg assignment. So this:

```python
        reach_list = [reach_func(df, rows=hours)]
```

And this as well:

```python
        reach_list = [
            reach_2_model(df, rows=hours),
            reach_3_model(df, rows=hours),
            reach_4_model(df, rows=hours),
            reach_5_model(df, rows=hours)
        ]
```